saleor/payment/utils.py

```diff
@@ -158,11 +176,43 @@ def gateway_authorize(payment: Payment, payment_token: str) -> Transaction:
     clean_authorize(payment)
 
     gateway, gateway_params = get_payment_gateway(payment.gateway)
-    with transaction.atomic():
-        txn, error = gateway.authorize(
-            payment=payment, payment_token=payment_token, **gateway_params)
+    try:
+        is_success, gateway_response, error = gateway.authorize(
+            payment=model_to_dict(payment), payment_token=payment_token,
```

Comment:
This ```model_to_dict``` should return information of a payment as more as possible. For example, some gateways may requires the shipping address, but this shipping address is not saved as a normal field of the Payment but in ```Payment.order.shipping_address.street_address_1|street_address_2|etc```. This method should make sure these kind of information is properly returned and then passed to gateways.